src/java.base/share/classes/java/nio/charset/Charset.java

```diff
@@ -374,23 +377,18 @@ public CharsetProvider next() {
                     return n;
                 }
 
-                public void remove() {
```

Comment:
`remove()` is specified to throw UOE unless overridden. So, removing this redundant code.